crates/telio-firewall/src/firewall.rs

```diff
@@ -696,7 +732,8 @@ impl StatefullFirewall {
             }
         };
         let flags = tcp_packet.get_flags();
-        Some((key, flags))
+        let seq = tcp_packet.get_sequence();
```

Comment:
Correct me if I am wrong, but I think this is not strictly correct, as the sequence number used in the TCP RST packet must be should be something like this:
```
TCP_RST_seq_num = Last_TCP_pkt_seq_num + TCP_payload_length
```

And the payload when sending ACK may very well be larger than 0 or 1.

Reply:
Yeah, you're right. It was overly simplified by me. Let me correct this one

Reply:
I've added the fix and expanded the test case for the client which uses an already synchronized sequence number so that  both paths are tested